src/odemis/driver/simcam.py

```diff
@@ -289,19 +329,69 @@ def _simulate(self):
         binning = self.binning.value
         res = self.resolution.value
         pxs_pos = self.translation.value
-        shape = self._img.shape
+
+        if self._stage:
+            pos = self._stage.position.value["x"], self._stage.position.value["y"]
+
+            exshift = numpy.array(self._orig_stage) - numpy.array(pos)
+            exshift = tuple(exshift)
+
+            # #convert to pxl
+            pxs = self._imgs[0].metadata[model.MD_PIXEL_SIZE]
+            X = exshift[0]/pxs[0]
+
+            Y = exshift[1] / pxs[1]
+            exs = (X,Y)
+
+            npos=numpy.array(pxs_pos) + numpy.array(exs)
+            pxs_pos=tuple(npos)
+
+            logging.debug('the position is %s',pxs_pos)
+
+        # Pick the image based on the current excitation wl
+
+        try:
+            current_inwl = self.getMetadata()[model.MD_IN_WL]   # (350e-9, 360e-9)
+            logging.debug('current_inwl %s',current_inwl)
+
+            current_inwl_c = util.fluo.get_center(current_inwl) # 355e-9
+
+            closest = float('inf')
+            _img = self._imgs[0]
+            for im in self._imgs:
+                c = util.fluo.get_center(im.metadata[model.MD_IN_WL])
+                if abs(current_inwl_c - c) < closest:
+                    _img = im
+                    closest = abs(current_inwl_c - c)
+
+            # _img = min(self._imgs, key=lambda im: abs(util.fluo.get_center(im.metadata[model.MD_IN_WL] - current_inwl_c)))
+
+            current_inwl = self.getMetadata().get(model.MD_IN_WL, None)
+            # logging.debug('current_inwl %s', current_inwl)
+            # logging.debug('im.metadata[model.MD_IN_WL] %s', self._imgs[0].metadata[model.MD_IN_WL])
+            # logging.debug('im.metadata[model.MD_IN_WL]1111111111111 %s', self._imgs[1].metadata[model.MD_IN_WL])
+
+        except KeyError:
+            _img = self._imgs[0]
+
+        shape = _img.shape
+
         center = (shape[1] / 2, shape[0] / 2)
-        lt = (center[0] + pxs_pos[0] - (res[0] / 2) * binning[0],
+
+        lt = (-center[0] - pxs_pos[0] - (res[0] / 2) * binning[0],
               center[1] + pxs_pos[1] - (res[1] / 2) * binning[1])
-        assert(lt[0] >= 0 and lt[1] >= 0)
+
+        # assert(lt[0] >= 0 and lt[1] >= 0)
```

Comment:
Why disabling the assert? If it's not valid anymore, remove it, otherwise, put it back.
BTW, how is it handled when the stage is causing the image area outside of the bounds (eg, <0 or >= res)? Does it wrap around? Is it clipped? I think there should be a comment explaining. Also, if the image returned is smaller than  the expected resolution, that's not good. It should always be the size requested by the user.
Maybe the use of `% res[0]` in the computation of `coord` would make it wrap properly.

Reply:
actually outside the bounds its black, this can cause problems if the acqusition goes over bounds as the detector does not receive any images, i will change it, what is the best option ?

Reply:
The modulo funciton, as proposed, should work fine, and it's very easy to implement.

Reply:
Please either put back the assert, if it's still correct, or entirely drop the line.